agent/consul/catalog_endpoint_test.go

```diff
@@ -2717,27 +2693,15 @@ func testACLFilterServer(t *testing.T) (dir, token string, srv *Server, codec rp
 	codec = rpcClient(t, srv)
 	testrpc.WaitForTestAgent(t, srv.RPC, "dc1", testrpc.WithToken("root"))
 
-	// Create a new token
-	arg := structs.ACLRequest{
-		Datacenter: "dc1",
-		Op:         structs.ACLSet,
-		ACL: structs.ACL{
-			Name: "User token",
-			Type: structs.ACLTokenTypeClient,
-			Rules: `
-service "foo" {
+	rules := `
+service_prefix "foo" {
```

Comment:
I suspect many of these `service "foo"` ones could be replaced by `service "foo" {} service "foo-sidecar-proxy" {}` instead of the foo prefix if you thought that was valuable.

Or you could switch these specific tests to use a "foo" ServiceIdentity (which would give it the right perms for doing a connect test).

Reply:
Sounds good, I'll do a pass through some of these to see if I can make the rules a bit more strict.